impl/src/main/java/org/ehcache/impl/serialization/TransientStateRepository.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.impl.serialization;
+
+import org.ehcache.impl.internal.concurrent.ConcurrentHashMap;
+import org.ehcache.spi.persistence.StateRepository;
+
+import java.io.ObjectStreamClass;
+import java.util.Map;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * TransientStateRepository
+ */
+class TransientStateRepository implements StateRepository {
+
+  @Override
+  public <K, V> ConcurrentMap<K, V> getPersistentConcurrentMap(String name) {
```

Comment:
These should get saved in a data structure so that repeated calls with the same name return the same map.